### PR TITLE
Depth First Search in Ruby

### LINKED ISSUE #936: Depth First Search in Ruby
Hello! 
I want to implement Depth First Search in Ruby.

### PR DESCRIPTION
**Pull request by Ibragim97**
Fixes #936 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Logic Documentation (Comments).
- [x] Sample Input / Output.
- [x] File names are correct.

#### Changes proposed in this pull request:

- Implemented Depth First Search

#### Languages Used:
- Ruby

#### Files Added:
- Depth_First_Search.rb

### CODE REVIEW

File: Depth_First_Search/Depth_First_Search.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
+
```
Comment (KanchanThareja) [inline Depth_First_Search/Depth_First_Search.rb]: delete this line

File: Depth_First_Search/Depth_First_Search.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            if used_verticies[adj_vertex] == false
+            then private_dfs(adj_vertex, used_verticies)
+            end
+            
+        }
+
+    end
+
+end 
+
```
Comment (KanchanThareja) [inline Depth_First_Search/Depth_First_Search.rb]: Add more comments to class graph

**Comment by Ibragim97** [comment]
Added some comments.

**Comment by jainaman224** [comment]
@KanchanThareja Please review.

**Comment by jainaman224** [comment]
@KanchanThareja Can you please review?

**Comment by KanchanThareja** [comment]
@jainaman224 apologies for the delay. I approve of the changes.

**Pull request merged**